code/extensions/Lumberjack.php

```diff
@@ -35,30 +35,48 @@ public function getExcludedSiteTreeClassNames() {
 	 * @param FieldList $fields
 	 */
 	public function updateCMSFields(FieldList $fields) {
-		$excluded = $this->owner->getExcludedSiteTreeClassNames();
-		if(!empty($excluded)) {
-			$pages = SiteTree::get()->filter(array(
-				'ParentID' => $this->owner->ID,
-				'ClassName' => $excluded
-			));
+		$childPages = $this->getLumberjackChildPages();
+
+		if($childPages !== null) {
 			$gridField = new GridField(
-				"ChildPages",
+				'ChildPages',
 				$this->getLumberjackTitle(),
-				$pages,
+				$childPages,
 				$this->getLumberjackGridFieldConfig()
 			);
 
 			$tab = new Tab('ChildPages', $this->getLumberjackTitle(), $gridField);
+
 			$fields->insertAfter($tab, 'Main');
 		}
 	}
 
+	/**
+	 * @return null|DataList
+	 */
+	protected function getLumberjackChildPages() {
+		if(method_exists($this->owner, 'getLumberjackChildPages')) {

```

Comment:
Try to use `->methodExists` if possible, since `method_exists` isn't aware of methods on extensions.

Also, do you think it could be generalised so that `Hierarchy` extension is all that's necessary?  We shouldn't need to bake in lumberjack-specific code to manage hierarchy when we already have an existing mechanism for this.

I'm not sure why we can't simply use `->AllChildren` (standard `Hierarchy` method to skip menu visibility filter) instead of the sitetree specific code that already exists. Worth having a look to make sure this isn't to resolve some special use case I might have missed (e.g. versioning).